packages/console/src/hooks/use-tenants.ts

```diff
@@ -0,0 +1,86 @@
+import { useLogto } from '@logto/react';
+import { type TenantInfo } from '@logto/schemas';
+import { type Optional, trySafe } from '@silverhand/essentials';
+import type ky from 'ky';
+import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+import { TenantsContext } from '@/contexts/TenantsProvider';
+
+import useSwrFetcher from './use-swr-fetcher';
+
+type KyInstance = typeof ky;
+
+const useTenants = (cloudApi: KyInstance) => {
```

Comment:
Can we call `useCloudApi()` inside this hook instead of passing it through param?

Reply:
We need cloudApi in the useTenants hook and most like also need this API in the place we call `useTenants` hook. Trying to keep them consistent. We can init the cloudApi in the hook and export it for other use cases.